openedx/core/djangoapps/content_tagging/api.py

```diff
@@ -175,6 +190,112 @@ def tag_content_object(
     return get_content_tags(object_key, taxonomy_id=taxonomy.id)
 
 
+def get_content_tags_for_object(
+    content_key: ContentKey,
+    include_children: bool,
+) -> tuple[TaggedContent, TaxonomyDict]:
+    """
+    Returns the object with the tags associated with it. If include_children is True, then it will also include
+    the children of the object and their tags.
+    """
+
+    def _get_object_tags(content_key: ContentKey, include_children: bool) -> QuerySet[ObjectTag]:
+        """
+        Return the tags for the object and its children using a single db query.
+        """
+        content_key_str = str(content_key)
+        if not include_children:
+            return ObjectTag.objects.filter(object_id=content_key_str).select_related("tag__taxonomy").all()
+
+        # We use a block_id_prefix (i.e. the modified course id) to get the tags for the children of the Content
+        # (course) in a single db query.
+        # ToDo: Add support for other content types (like LibraryContent and LibraryBlock)
+        if isinstance(content_key, UsageKey):
+            course_key_str = str(content_key.course_key)
+            block_id_prefix = course_key_str.replace("course-v1:", "block-v1:", 1)
+        elif isinstance(content_key, CourseKey):
+            course_key_str = str(content_key)
+            block_id_prefix = str(content_key).replace("course-v1:", "block-v1:", 1)
+        else:
+            raise NotImplementedError(f"Invalid content_key: {type(content_key)} -> {content_key}")
+
+        return ObjectTag.objects.filter(Q(object_id__startswith=block_id_prefix) | Q(object_id=course_key_str)) \
```

Comment:
Need to exclude "deleted" tags here too (object tags with tag=None or taxonomy=None).

This also guarantees that calling `x.tag.taxonomy` below doesn't fail where `x.tag` is None.

Reply:
Done: 9726d6d23f9edae217a28ec7fab466d736960a4b